exercises/algorithms/src/test/java/com/cbfacademy/BinarySearchTest.java

```diff
@@ -0,0 +1,77 @@
+package com.cbfacademy;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.cbfacademy.search.BinarySearch;
+import com.cbfacademy.search.Search;
+
+@DisplayName(value = "BinarySearch Exercise")
+public class BinarySearchTest {
+    Search binarySearch;
+    int[] testArray = new int[50];
+
+    @BeforeEach
+    public void setUp() {
+        binarySearch = new BinarySearch();
+        Arrays.setAll(testArray, position -> position + 1);
+    }
+
+    @Test
+    @DisplayName("returns this unit's name")
+    public void returnUnitName() {
+        String name = binarySearch.getName();
+
+        assertThat(name, is("Binary Search Algorithm Exercise"));
+    }
+
+    @Test
+    @DisplayName("search returns the index of the target value when it is in the middle of the array")
+    public void returnsIndexWhenTargetIsMidPointInArray() {
+        int result = binarySearch.search(testArray, 25);
+
+        assertTrue(result == 24);
+    }
+
+    @Test
+    @DisplayName("search returns the index of the target value when it is in the upper end of the array")
+    public void returnsIndexWhenTargetUpperEndOfArray() {
+        int result = binarySearch.search(testArray, 48);
+
+        assertTrue(result == 47);
+    }
+
+    @Test
+    @DisplayName("search returns the index of the target value when it is in the lower end of the array")
+    public void returnsIndexWhenTargetIsLowerEndOfArray() {
+        int result = binarySearch.search(testArray, 6);
+
+        assertTrue(result == 5);
+    }
```

Comment:
These can be combined into a single parameterised test, e.g.

```java
    @ParameterizedTest
    @MethodSource
    @DisplayName("search() method returns index of target value")
    public void search_ReturnsIndexOfTargetValue(int target) {
        int index = binarySearch.search(testArray, target);

        assertTrue(index == target - 1);
    }

    private static search_ReturnsIndexOfTargetValue() {
        int maxIndex = testArray.length - 1;
        int[] targetIndices = { 0, 1, maxIndex / 4, maxIndex / 3, maxIndex / 2, maxIndex * 2 / 3, maxIndex * 3 / 4, maxIndex - 1, maxIndex}; // Test for values at relative positions rather than hard-coded indices
        List<Arguments> searchArguments = new ArrayList<>();

        for (int index : targetIndices) {
            searchArguments.add(arguments(testArray[index]));
        }

        return searchArguments.stream();
    }
```